Fix modded faction icons and tooltip colors in create unit dialog

lL1l1 opened the pull request:
## Issue
The create unit dialog was missing the icon for nomads units and the tooltip when you mouse over a unit was the wrong color.

## Description of the proposed changes
- Get the icons from the factions table.
- Get the tooltip text color from the skin for the unit's faction.
- Annotate the code in `factions.lua`.

## Testing done on the proposed changes
Tested with the Nomads repo and a dev environment set up. The icon is now shown and the changed text color was not hard to read on the tooltip background for any faction.

<details> <summary> Before: </summary>

![{3F981036-CBE2-45AA-948B-9683EB0E033A}](https://github.com/user-attachments/assets/320b37f0-9538-402f-b5f9-c95d21f2140d)
![{2EBDBC33-5DF7-4B49-BE55-F04EADEF00DC}](https://github.com/user-attachments/assets/f63ef1cd-2e0c-4c34-8115-e6af00a8c246)
![{DAE495DE-0208-41D4-B750-A9EC8133F132}](https://github.com/user-attachments/assets/e363fa7d-5515-42e2-99eb-f8f5f331c828)
![{4B14E159-7775-4F7A-A5B7-790C396842FD}](https://github.com/user-attachments/assets/9d011464-72f0-4058-aa6b-9b432542e39d)
![{AA73D91F-D7D6-4DAC-ABE4-2019F3CB23A7}](https://github.com/user-attachments/assets/c17758f8-d2c3-434c-ab4e-4c84caa9ece6)


</details>

<details> <summary> After: </summary>

![{95F49CC1-29D0-4198-8C16-9108A3400AC0}](https://github.com/user-attachments/assets/6fb721f0-6747-4b4b-9227-e126848c430e)
![{A7BD2A32-ACE3-424F-900E-BD8F2B807D0B}](https://github.com/user-attachments/assets/2284a058-c152-4e43-9024-ab54efdfd87e)
![{B8C4E7F4-FE97-4675-A161-9096E7DEA1DE}](https://github.com/user-attachments/assets/5f4a9d3f-471e-4941-87f0-3afd5f4bf525)
![{57023669-8117-4254-ABC5-06A86318D104}](https://github.com/user-attachments/assets/1a9ec6f5-e971-4a3a-aca9-71b4eef5e9eb)
![{FE47922B-9AF7-4D4D-A8C8-4DDB7B68A946}](https://github.com/user-attachments/assets/e67abf70-7453-44ec-a3ff-39d2104bee95)

</details>

## Checklist
- [x] Changes are annotated, including comments where useful
- [x] Changes are documented in a changelog snippet according to the [guidelines](https://faforever.github.io/fa/development/changelog#format-of-a-snippet).
- [x] Request 2-3 reviewers from the [list of reviewers and their areas of knowledge](https://github.com/FAForever/fa/blob/deploy/fafdevelop/CONTRIBUTING.md#reviewers).

lL1l1 committed:
Fix create unit dialog's unit icon/tooltip faction visuals

Most obvious was that the Nomads icon was missing.
This also changes the faction text color used in the tooltips, but this makes it all consistent and doesn't seem difficult to read for me with the tooltip background.

lL1l1 committed:
Annotate factions.lua

lL1l1 committed:
Fix unit bp FactionName type

lL1l1 committed:
Format `factions.lua`

lL1l1 committed:
Add regions

lL1l1 committed:
Create fix.6872.md

4z0t commented on lua/ui/dialogs/createunit.logic.lua:
```diff
@@ -115,12 +115,17 @@ end
-    { color = 'ffff0000', name = 'CYBRAN', icon = UIUtil.UIFile(UIUtil.GetFactionIcon(2)) },
-    { color = 'FFFFBF00', name = 'SERAPHIM', icon = UIUtil.UIFile(UIUtil.GetFactionIcon(3)) },
-}
+local FactionData = {}
+local Factions = import('/lua/factions.lua').GetFactions()
+local skins = import("/lua/skins/skins.lua").skins
+---@param faction FactionData
+for _, faction in Factions do
+    table.insert(FactionData, {
+        name = faction.Category,
+        icon = UIUtil.UIFile(faction.Icon),
+        color = skins[faction.Key].factionTextColor
+    })
+end
```
I'm against any logic that happens during module loading (any statements including `if`, `for` or `while` as well as function calls that are not common). It is prone to break or is unreliable.

lL1l1 replied:
What do you mean by "function calls that are not common"? Otherwise I agree, it's easier to mod, debug, and doesn't run logic that might fail at inopportune moments.

4z0t replied:
Functions like `import` and `Class` for example

lL1l1 committed:
Wrap the data creation in a function

lL1l1 committed:
Add annotations

lL1l1 committed:
Do imports before instantiating the table

if the import fails then there would be an opportunity to fix the import with a hot-reload and then create the table with data.

lL1l1 committed:
Remove param annotation

should fix intellisense instead

lL1l1 committed:
Use `_` for unused variable in for loop
